recipes/AMI/Diarization/experiment.py

```diff
@@ -236,10 +236,10 @@ def diarize_dataset(full_meta, split_type, n_lambdas, pval, n_neighbors=10):
         diary_set_loader = dataio_prep(params, meta_per_rec_file)
 
         # Putting modules on the device.
-        params["compute_features"].to(params["device"])
-        params["mean_var_norm"].to(params["device"])
-        params["embedding_model"].to(params["device"])
-        params["mean_var_norm_emb"].to(params["device"])
+        params["compute_features"].to(run_opts["device"])
+        params["mean_var_norm"].to(run_opts["device"])
```

Comment:
How does DDP handle this?

DDP testing would be part of a follow-up PR. Here, it's single GPU testing only (or CPU if wanted).

Reply:
I removed the device field from params, since it is handled in the Brain class. There, device is obtained from run_opts - this recipe however is not using the sb.Brain class at all. As every recipe, it has `params_file, run_opts, overrides = sb.core.parse_arguments(sys.argv[1:])` through which `run_opts["device"]` is set. Hparams defined it before to be only `cuda:0` which has two limitations (no cpu & fixed GPU ID)